### PR TITLE
fix(api): is_first_launch not cleared when setup wizard completes without prior profile switch

### LINKED ISSUE #291: fix(api): is_first_launch not cleared when setup wizard completes without prior profile switch
## Summary

`is_first_launch: Arc<AtomicBool>` is only cleared by `compare_exchange` in `profile_switch.rs`. The setup wizard handler (`services/api/src/auth/mod.rs:256`) writes the `active_profile` file to disk but does NOT call `compare_exchange`, so `is_first_launch` remains `true` for the lifetime of that server process even after setup completes.

The Session 3 welcome screen UX always routes through `POST /api/profile/switch` before reaching setup, so the `compare_exchange` fires and the flag is cleared correctly. The bug only manifests if the setup API is called without going through the welcome screen (e.g., scripted onboarding, direct API use, or unexpected navigation).

## Acceptance Criteria
- [ ] After setup wizard completes, `GET /api/setup-status` returns `is_first_launch: false` in the same server process
- [ ] Fix: add `state.is_first_launch.compare_exchange(true, false, Ordering::AcqRel, Ordering::Relaxed)` in the setup wizard handler after writing `active_profile` to disk
- [ ] BDD scenario: "After setup wizard completion without prior profile switch, is_first_launch is false"

## Context

Flagged during Session 2 review (PR #287) by test analyzer. Pre-existing before Session 3 ships — Session 3 frontend avoids this via the welcome → profile switch → setup flow, but the edge case is real and should be closed.

### PR DESCRIPTION
**Pull request by cmbays**
## Summary

- `is_first_launch: Arc<AtomicBool>` was only cleared by `profile_switch.rs` via `compare_exchange`. The setup wizard handler wrote `active_profile` to disk and updated in-memory state but never cleared the flag, leaving `is_first_launch: true` for the process lifetime when setup completed without going through the welcome screen's profile switch path (scripted onboarding, direct API use).
- Adds `compare_exchange(true, false, AcqRel, Relaxed)` in the setup handler after `active_profile` is written — mirrors the identical pattern already in `profile_switch.rs:210`.
- Adds two BDD scenarios: one asserting `is_first_launch: false` after successful setup via the real HTTP API, one asserting it stays `true` after a rejected setup attempt (pins the invariant).

## Test Plan

- [x] `moon run api:test` — 296/296 pass
- [x] BDD: "Setup wizard completion clears is_first_launch without a prior profile switch" — GREEN
- [x] BDD: "Failed setup wizard does not clear is_first_launch" — GREEN (invariant pinned)

Closes #291

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Resolved an issue where the setup wizard status was not properly reflected as complete after completion in certain scenarios.

* **Tests**
  * Added test coverage for setup wizard completion and setup status verification.

### CODE REVIEW

File: services/api/tests/features/setup_wizard.feature [hunk 1]
```diff
@@ -61,3 +61,14 @@ Feature: Setup Wizard
     When someone submits the setup wizard with an incorrect token
     Then the request is rejected
     And no user account is created
+
+  Scenario: Setup wizard completion clears is_first_launch without a prior profile switch
+    Given a freshly started server with no users
+    When the shop owner completes the setup wizard via the HTTP API
```
[the remark this answers is not in the document]
Comment (cmbays) [reply services/api/tests/features/setup_wizard.feature:67]: Not a defect. ApiWorld::new() (tests/bdd_world/mod.rs line 87) calls build_app_with_shutdown and stores the real setup token into w.setup_token. The 'Given a valid setup token' step (auth_steps.rs line 28) overwrites it with "test-token-placeholder", which causes a 401 on the actual HTTP POST. The scenario is intentionally written without that step so it uses the real token that ApiWorld::new() already provides. No change needed.
